File: schemas/crm/drizzle/companies.ts
// companies: organizations tracked in the CRM with industry and revenue data.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, numeric, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "./users";

export const companies = pgTable(
  "companies",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    domain: text("domain").unique(),
    industry: text("industry"),
    employeeCount: integer("employee_count"),
    annualRevenue: numeric("annual_revenue"),
    phone: text("phone"),
    addressStreet: text("address_street"),
    addressCity: text("address_city"),
    addressState: text("address_state"),
    addressCountry: text("address_country"),
    addressZip: text("address_zip"),
    website: text("website"),
    description: text("description"),
    ownerId: uuid("owner_id").references(() => users.id, { onDelete: "set null" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_companies_owner_id").on(table.ownerId),
    index("idx_companies_industry").on(table.industry),
  ]
);

File: schemas/crm/spacetimedb/companies.rs
// companies: Organizations and businesses associated with contacts and deals.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = companies, public)]
pub struct Company {
    #[primary_key]
    pub id: String, // UUID
    pub name: String,
    #[unique]
    pub domain: Option<String>,
    #[index(btree)]
    pub industry: Option<String>,
    pub employee_count: Option<i32>,
    pub annual_revenue: Option<f64>,
    pub phone: Option<String>,
    pub address_street: Option<String>,
    pub address_city: Option<String>,
    pub address_state: Option<String>,
    pub address_country: Option<String>,
    pub address_zip: Option<String>,
    pub website: Option<String>,
    pub description: Option<String>,
    #[index(btree)]
    pub owner_id: Option<String>, // UUID — FK → users.id (set null)
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
